app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Admin::VideosController < AdminsController
+
+  def new
+    @video = Video.new
+  end
+
+  def create
+    @video = Video.new video_params
+    if @video.save
+      flash[:success] = "The video #{@video.title} has been created."
+      redirect_to new_admin_video_path
+    else
+      flash[:danger] = 'The video could not be created.'
```

Comment:
Make sure you use `flash.now` when you use `render`.  `flash` will persist through the next page load as well.